src/factories/flowRunArtifact.ts

```diff
@@ -0,0 +1,46 @@
+import { Container } from 'pixi.js'
+import { DEFAULT_ROOT_ARTIFACT_BOTTOM_OFFSET } from '@/consts'
+import { artifactFactory } from '@/factories/artifact'
+import { Artifact } from '@/models'
+import { waitForApplication, waitForViewport } from '@/objects'
+import { emitter } from '@/objects/events'
+import { waitForScale } from '@/objects/scale'
+import { layout } from '@/objects/settings'
+
+export type FlowRunArtifactFactory = Awaited<ReturnType<typeof flowRunArtifactFactory>>
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function flowRunArtifactFactory(artifact: Artifact) {
+  const application = await waitForApplication()
+  const viewport = await waitForViewport()
+  let scale = await waitForScale()
+
+  const { element, render: renderArtifact } = await artifactFactory(artifact, true)
+
+  emitter.on('scaleUpdated', updated => scale = updated)
+
+  application.ticker.add(() => {
+    updatePosition()
```

Comment:
Same here. Be nice to not constantly  be setting the position to the same value if the viewport hasn't actually changed.